### PR TITLE
SDIT-2110 Handle merges for casenotes

### PR DESCRIPTION
**Pull request by steverendell**
prisoner from-to merge

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingService.kt [hunk 1]
```diff
@@ -171,6 +171,20 @@ class CaseNoteMappingService(
       .map { it.toDto() }
       .let { AllPrisonerCaseNoteMappingsDto(it) }
 
+  @Transactional
+  suspend fun updateMappingsByNomisId(oldOffenderNo: String, newOffenderNo: String) {
+    val count = repository.updateOffenderNo(oldOffenderNo, newOffenderNo)
+    telemetryClient.trackEvent(
+      "nonAssociation-mapping-merged",
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingService.kt:178]: is this a non association?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt [hunk 2]
```diff
@@ -477,6 +478,28 @@ class CaseNotesMappingResource(private val mappingService: CaseNoteMappingServic
     @RequestParam(name = "onlyMigrated", required = false, defaultValue = "false") onlyMigrated: Boolean,
   ) = mappingService.deleteMappings(onlyMigrated)
 
+  @PutMapping("/merge/from/{oldOffenderNo}/to/{newOffenderNo}")
+  @Operation(
+    summary = "Replaces all occurrences of the 'from' id with the 'to' id in the mapping table",
+    description = "Used for update after a prisoner number merge. Requires role ROLE_NOMIS_CASENOTES",
+    responses = [
+      ApiResponse(responseCode = "200", description = "Replacement made, or not present in table"),
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt:486]: seems a bit weird to return a `200` without a body?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt [hunk 3]
```diff
@@ -477,6 +478,28 @@ class CaseNotesMappingResource(private val mappingService: CaseNoteMappingServic
...
+    ],
+  )
+  suspend fun updateMappingsByNomisId(
+    @Schema(description = "Old prisoner number to replace", example = "A3456KM", required = true)
+    @PathVariable
+    oldOffenderNo: String,
+    @Schema(description = "New prisoner number to use", example = "A3457LZ", required = true)
+    @PathVariable
+    newOffenderNo: String,
+  ) = mappingService.updateMappingsByNomisId(oldOffenderNo, newOffenderNo)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt:501]: this is actually a `void` method?  Just wondering what is being returned

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingService.kt (hunk 1 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingService.kt:178]: whoah nope ! Good spot 🤦

Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingService.kt:178]: will sort in next pr which is coming imminently

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt (hunk 2 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt:486]: I thought that was ok for a put. Ill see what other endpoints are doing

Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt:486]: Ill give the function a block body

**Pull request merged**